deepfence_server/reporters/scan/scan_result_actions.go

```diff
@@ -239,54 +240,60 @@ func StopCloudComplianceScan(ctx context.Context, scanType, scanId string) error
 	defer tx.Close()
 
 	query := `MATCH (n:%s) -[:SCANNED]-> ()
-        WHERE n.node_id = $scan_id
+        WHERE n.node_id = $scanid
         AND n.status = $in_progress
         SET n.status=$cancel_pending`
 
-	if _, err = tx.Run(fmt.Sprintf(query, scanType),
-		map[string]interface{}{
-			"scan_id":        scanId,
-			"in_progress":    utils.SCAN_STATUS_INPROGRESS,
-			"cancel_pending": utils.SCAN_STATUS_CANCEL_PENDING,
-		}); err != nil {
-		log.Error().Msgf("StopCloudComplianceScan: Error in setting the state in neo4j: %v", err)
-		return err
+	for _, scanid := range scanIds {
+		if _, err = tx.Run(fmt.Sprintf(query, scanType),
+			map[string]interface{}{
+				"scanid":         scanid,
+				"in_progress":    utils.SCAN_STATUS_INPROGRESS,
+				"cancel_pending": utils.SCAN_STATUS_CANCEL_PENDING,
+			}); err != nil {
+			log.Error().Msgf("StopCloudComplianceScan: Error in setting the state in neo4j: %v", err)
+			return err
+		}
 	}
 
 	return tx.Commit()
 }
 
-func StopScan(ctx context.Context, scanType, scanId string) error {
-
+func StopScan(ctx context.Context, scanType string, scanIds []string) error {
 	driver, err := directory.Neo4jClient(ctx)
 	if err != nil {
 		return err
 	}
 	session := driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeWrite})
 	defer session.Close()
-
 	tx, err := session.BeginTransaction(neo4j.WithTxTimeout(15 * time.Second))
 	if err != nil {
 		return err
 	}
 	defer tx.Close()
 
-	query := `MATCH (n:%s) -[:SCANNED]-> ()
-        WHERE n.node_id = $scan_id
-        AND n.status IN [$starting, $in_progress]
-        SET n.status=$cancel_pending`
-
-	if _, err = tx.Run(fmt.Sprintf(query, scanType),
-		map[string]interface{}{
-			"scan_id":        scanId,
-			"starting":       utils.SCAN_STATUS_STARTING,
-			"in_progress":    utils.SCAN_STATUS_INPROGRESS,
-			"cancel_pending": utils.SCAN_STATUS_CANCEL_PENDING,
-		}); err != nil {
-		log.Error().Msgf("StopScan: Error in setting the state in neo4j: %v", err)
-		return err
+	nodeStatusField := ingestersUtil.ScanStatusField[utils.Neo4jScanType(scanType)]
+	latestScanIdField := ingestersUtil.LatestScanIdField[utils.Neo4jScanType(scanType)]
+	query := `MATCH (n:%s) -[:SCANNED]-> (m)
+			WHERE n.node_id = $scanid
+			AND n.status IN [$starting, $in_progress]
+			WITH n,m                
+			SET n.status= CASE WHEN n.status = $starting THEN $cancelled 
+			ELSE $cancel_pending END, 
+			m.%s=n.status, m.%s=$scanid`
+
+	for _, scanid := range scanIds {
+		if _, err = tx.Run(fmt.Sprintf(query, scanType, nodeStatusField, latestScanIdField),
```

Comment:
Let's do the `Sprintf` outside the loop to avoid unnecessary allocations

Reply:
Done.